File: experiments/eval/benchmarks/in-house/fixtures/dispatch/metrics.go
package dispatch

// MetricsBuffer is the decoy. It has a Send method and is therefore
// indistinguishable from a Notifier to any name-based matcher, but it is never
// used as one: FlushMetrics calls it directly on the concrete type.
//
// It exists to make precision measurable. An arm that answers "who dispatches
// through Notifier?" with FlushMetrics has confused a shared method name for a
// shared interface — the failure mode both text search and name-only interface
// resolution are prone to.
type MetricsBuffer struct {
	pending int
}

func (m *MetricsBuffer) Send(msg string) error {
	_ = m.pending
	_ = msg
	return nil
}

func FlushMetrics(m *MetricsBuffer, msg string) error {
	return m.Send(msg)
}

File: experiments/eval/benchmarks/in-house/fixtures/dispatch-ts/metrics.ts
// MetricsBuffer is the decoy. It has a send method with a matching signature,
// so a structural rule would call it a Notifier — but it declares no
// `implements`, and nothing ever uses it as one.
//
// This is where TypeScript differs from Go: the declared rule excludes it, so
// the decoy that costs every arm precision on the Go fixture costs nothing here.
export class MetricsBuffer {
  send(msg: string): void {
    void msg;
  }
}

export function flushMetrics(m: MetricsBuffer, msg: string): void {
  m.send(msg);
}
